service/dap/server.go

```diff
@@ -860,7 +860,21 @@ func (s *Server) convertVariable(v api.Variable) (value string, variablesReferen
 			}
 			value = fmt.Sprintf("%q", vvalue)
 		}
-	default: // Structs, scalars
+	case reflect.Chan:
+		if len(v.Children) == 0 {
+			value = "nil <" + v.Type + ">"
+		} else {
+			value = "<" + v.Type + ">"
+			variablesReference = s.variableHandles.create(v)
+		}
+	case reflect.Interface:
+		if v.Children[0].Kind == reflect.Invalid && v.Children[0].Addr == 0 {
```

Comment:
This is an improvement over what vscode-go currently does.
I replaced:
<img src="https://user-images.githubusercontent.com/51177946/88309048-ab6d2500-ccc2-11ea-9b70-0fc98f6121ae.png" alt="alt text" height="50">
with
<img src="https://user-images.githubusercontent.com/51177946/88309224-deafb400-ccc2-11ea-8968-41cc6d847ba2.png" alt="alt text" height="25">
This is consistent with how pointers are handled and saves users unnecessary clicks.
<img src="https://user-images.githubusercontent.com/51177946/88309524-4239e180-ccc3-11ea-935f-90be4d274c44.png" alt="alt text" height="75">